pools/usecase/pools_usecase.go

```diff
@@ -540,7 +546,78 @@ func (p *poolsUseCase) CalcExitCFMMPool(poolID uint64, exitingShares osmomath.In
 
 	// fine to pass empty context as no data is mutated
 	exitFee := pool.GetExitFee(sdk.Context{})
-	return pool.CalcExitPoolCoinsFromShares(sdk.Context{}, exitingShares, exitFee)
+
+	return calcExitPool(sdk.Context{}, pool, exitingSharesIn, exitFee)
+}
+
+// errMsgFormatSharesLargerThanMax is the error message format for when the exiting shares are larger than the max allowed.
+const errMsgFormatSharesLargerThanMax = "cannot exit all shares in a pool. Attempted to exit %f shares, max allowed is %f"
+
+// calcExitPool is a helper function to calculate the exit pool.
+// It is a direct port of the CalcExitPool function from the Gamm module with some modifications
+// to optimize the calculation for performance.
+func calcExitPool(ctx sdk.Context, pool types.CFMMPoolI, exitingSharesIn osmomath.Int, exitFee osmomath.Dec) (sdk.Coins, error) {
+	totalShares, err := strconv.ParseFloat(pool.GetTotalShares().String(), 64)
+	if err != nil {
+		return sdk.Coins{}, err
+	}
+
+	exitingShares, err := strconv.ParseFloat(exitingSharesIn.String(), 64)
+	if err != nil {
+		return sdk.Coins{}, err
+	}
```

Comment:
I initially thought that we could convert to int64 and then to float for perf reasons but realizing that we would easily overflow the int